tests/Api/HttpRestClientTest.php

```diff
@@ -0,0 +1,556 @@
+
+        $this->expectRequest(
+            'POST',
+            '/consumers/dolor-sit-amet/instances/custom-consumer-name/subscription',
+            '{"topic_pattern":"lorem-*"}',
+            $this->response->withStatus(Http::NO_CONTENT)
+        );
+
+        $this->client->consumerSubscribe($consumer, Subscription::pattern('lorem-*'));
+    }
+
+    public function testGetConsumerSubscribedTopics() : void
+    {
+        $consumer = new Consumer();
+        $consumer->baseUri = 'http://api/consumers/dolor-sit-amet/instances/custom-consumer-name';
+
+        $this->expectRequest(
+            'GET',
+            '/consumers/dolor-sit-amet/instances/custom-consumer-name/subscription',
+            null,
+            $this->responseWithBody('{"topics":["lorem-ipsum"]}')
+        );
+
+        $topics = $this->client->getConsumerSubscribedTopics($consumer);
+
+        self::assertCount(1, $topics);
+        self::assertSame('lorem-ipsum', $topics[0]);
+    }
+
+    /**
+     * @doesNotPerformAssertions
+     */
+    public function testConsumerUnsubscribe() : void
+    {
+        $consumer = new Consumer();
+        $consumer->baseUri = 'http://api/consumers/dolor-sit-amet/instances/custom-consumer-name';
+
+        $this->expectRequest(
+            'DELETE',
+            '/consumers/dolor-sit-amet/instances/custom-consumer-name/subscription',
+            null,
+            $this->response->withStatus(Http::NO_CONTENT)
+        );
+
+        $this->client->consumerUnsubscribe($consumer);
+    }
+
+    /**
+     * @doesNotPerformAssertions
+     */
+    public function testConsumerAssignPartitions() : void
+    {
+        $consumer = new Consumer();
+        $consumer->baseUri = 'http://api/consumers/dolor-sit-amet/instances/custom-consumer-name';
+
+        $this->expectRequest(
+            'POST',
+            '/consumers/dolor-sit-amet/instances/custom-consumer-name/assignments',
+            '{"partitions":[{"topic":"lorem-ipsum","partition":0}]}',
+            $this->response->withStatus(Http::NO_CONTENT)
+        );
+
+        $this->client->consumerAssignPartitions($consumer, [new Partition('lorem-ipsum', 0)]);
+    }
+
+    public function testGetConsumerAssignedPartitions() : void
+    {
+        $consumer = new Consumer();
+        $consumer->baseUri = 'http://api/consumers/dolor-sit-amet/instances/custom-consumer-name';
+
+        $this->expectRequest(
+            'GET',
+            '/consumers/dolor-sit-amet/instances/custom-consumer-name/assignments',
+            null,
+            $this->responseWithBody('{"partitions":[{"topic":"lorem-ipsum","partition":0}]}')
+        );
+
+        $partitions = $this->client->getConsumerAssignedPartitions($consumer);
+
+        self::assertCount(1, $partitions);
+        self::assertSame('lorem-ipsum', $partitions[0]->topic);
+        self::assertSame(0, $partitions[0]->partition);
+    }
+
+    /**
+     * @doesNotPerformAssertions
```

Comment:
I'd recommend self::assertTrue(true) at the end.

Context: https://github.com/sebastianbergmann/phpunit/issues/3016